dex/src/state.rs

```diff
@@ -2544,43 +2685,9 @@ impl State {
             }
         };
 
-        if deposit_amount != 0 {
```

Comment:
There's some subtlety here. This code was moved below to facilitate the case where the open orders account *authority* is the open orders account itself (i.e. a PDA). In this case, `invoke_spl_token` will try to borrow the account info refcell, which would cause an error, as there would be two borrows while one of them is mutable.

To fix this, we move this code below, drop the open_orders mutable borrow before executing it.